src/gui/editlocallyjob.cpp

```diff
@@ -118,42 +118,65 @@ void EditLocallyJob::remoteTokenCheckResultReceived(const int statusCode)
         return;
     }
 
-    proceedWithSetup();
+    _folderForFile = findFolderForFile(_relPath, _userId);
+
+    if (!_folderForFile) {
+        showError(tr("Could not find a file for local editing. Make sure it is not excluded via selective sync."), _relPath);
+        return;
+    }
+
+    _relativePathToRemoteRoot = getRelativePathToRemoteRootForFile();
+
+    if (_relativePathToRemoteRoot.isEmpty()) {
+        qCWarning(lcEditLocallyJob) << "_relativePathToRemoteRoot is empty for" << _relPath;
+        showError(tr("Could not find a file for local editing. Make sure it is not excluded via selective sync."), _relPath);
+        return;
+    }
+
+    _relPathParent = getRelativePathParent();
+
+    if (_relPathParent.isEmpty()) {
+        showError(tr("Could not find a file for local editing. Make sure it is not excluded via selective sync."), _relPath);
+        return;
+    }
+
+    if (_relPathParent == QStringLiteral("/")) {
+        proceedWithSetup();
+        return;
+    }
+
+    fetchRemoteFileParentInfo();
```

Comment:
I can see the use of the new `getRelativePathParent` 🙂 I still don't think we should have this much code unrelated to the token check result processing in `remoteTokenCheckResultReceived`, though, this is slightly misleading...

I would either move all of this to a new method that then itself calls `proceedWithSetup`, or move this to `proceedWithSetup` itself

Reply:
I suspect the former solution will be easier than the latter due to the job that needs to be run in `fetchRemoteFileParentInfo`

Reply:
FWIW I also don't think I designed this very well when I added a call to `proceedWithSetup` in `remoteTokenCheckResultReceived`, so if there is a better design to be put here then even better

Reply:
@claucambra Besides having a state-machine, I don't see another solution, so, let's keep that `proceedWithSetup` and I am gonna move the unrelated part out of `remoteTokenCheckResultReceived`